addons/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/internal/discovery/PLCDiscoveryService.java

```diff
@@ -0,0 +1,164 @@
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.model.script.actions.Ping;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PLCDiscoveryService} is responsible for discovering devices on
+ * the current Network. It uses every Network Interface which is connected to a network.
+ * Based on network binding discovery service.
+ *
+ * @author Alexander Falkenstern - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class)
+public class PLCDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(PLCDiscoveryService.class);
+
+    // Bridge config properties
+    private static final String LOGO_HOST = "address";
+    private static final int LOGO_PORT = 102;
+
+    private static final int DISCOVERY_TIMEOUT = 30;
+    private static final Set<ThingTypeUID> THING_TYPES_UIDS = Collections.singleton(THING_TYPE_DEVICE);
+
+    private static final int CONNECTION_TIMEOUT = 500;
+    private Set<String> addresses = new TreeSet<>();
+
+    private @Nullable ExecutorService executor;
+
+    private class Runner implements Runnable {
+        private final ReentrantLock lock = new ReentrantLock();
+        private String host;
+
+        public Runner(final String address) {
+            Objects.requireNonNull(address, "IP may not be null");
+            this.host = address;
+        }
+
+        @Override
+        public void run() {
+            try {
+                if (Ping.checkVitality(host, LOGO_PORT, CONNECTION_TIMEOUT)) {
+                    logger.info("LOGO! device found at: {}.", host);
+
+                    ThingUID thingUID = new ThingUID(THING_TYPE_DEVICE, host.replace('.', '_'));
+                    DiscoveryResultBuilder builder = DiscoveryResultBuilder.create(thingUID);
+                    builder.withProperty(LOGO_HOST, host);
+                    builder.withLabel(host);
+
+                    lock.lock();
+                    try {
+                        thingDiscovered(builder.build());
+                    } finally {
+                        lock.unlock();
+                    }
+                }
+            } catch (IOException exception) {
+                logger.debug("LOGO! device not found at: {}.", host);
+            }
+        }
+    }
+
+    /**
+     * Constructor.
+     */
+    public PLCDiscoveryService() {
+        super(THING_TYPES_UIDS, DISCOVERY_TIMEOUT);
+    }
+
+    @Override
+    protected void startScan() {
+        stopScan();
+
+        try {
+            Enumeration<NetworkInterface> devices = NetworkInterface.getNetworkInterfaces();
```

Comment:
Instead of doing this on your own, you should use the `NetUtil.getAllInterfaceAddresses()` method

Reply:
Please analyse why this is so. If you [look at the code](https://github.com/eclipse/smarthome/blob/cc5b310af8c3183c3aeb88fdb14e063bfc6af89d/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java#L362-L403), you will see that it is doing more or less the exact same thing as you, but it also filters interfaces which are not up (which is imho a bug in your code).

Reply:
Ok. The main difference is, i fetch all adresses contained by a given IP network and not by interface: `addresses.addAll(Arrays.asList(utilities.getInfo().getAllAddresses()));` Lack of "isUp" can produce more IP's to scan. Added it to implementation. Thanks for the hint.